dowhy/do_why.py

```diff
@@ -158,6 +158,49 @@ def estimate_effect(self, identified_estimand, method_name=None,
             )
         return estimate
 
+    def estimate_outcome(self, x, identified_estimand, method_name=None,
```

Comment:
Like the idea of a "do" function that works like the do operator. But looks like it is not implemented yet. Maybe you can submit the g-formula file separately from these edits, which may come in a future commit?